sfge/src/main/java/com/salesforce/rules/unusedmethod/RuleStateTracker.java

```diff
@@ -85,9 +149,27 @@ List<InvocableWithParametersVertex> getMethodCallsByDefiningType(String defining
         List<InvocableWithParametersVertex> methodCalls =
                 SFVertexFactory.loadVertices(
                         g, g.V().where(H.has(targetLabels, Schema.DEFINING_TYPE, definingType)));
-        // Cache and return the results.
-        this.methodCallsByDefiningType.put(definingType, methodCalls);
-        return methodCalls;
+        // Sort the results by type and cache them appropriately.
+        List<MethodCallExpressionVertex> methodCallExpressions = new ArrayList<>();
+        List<ThisMethodCallExpressionVertex> thisMethodCallExpressions = new ArrayList<>();
+        List<SuperMethodCallExpressionVertex> superMethodCallExpressions = new ArrayList<>();
+        for (InvocableWithParametersVertex invocable : methodCalls) {
+            if (invocable instanceof MethodCallExpressionVertex) {
+                methodCallExpressions.add((MethodCallExpressionVertex) invocable);
+            } else if (invocable instanceof ThisMethodCallExpressionVertex) {
+                thisMethodCallExpressions.add((ThisMethodCallExpressionVertex) invocable);
+            } else if (invocable instanceof SuperMethodCallExpressionVertex) {
+                superMethodCallExpressions.add((SuperMethodCallExpressionVertex) invocable);
+            } else {
+                throw new UnexpectedException(
```

Comment:
Can we make this a TodoException? Sounds like something that could possibly happen but we're not sure when.